torchdata/datapipes/iter/load/huggingface.py

```diff
@@ -62,19 +74,14 @@ def __init__(
             )
 
         self.dataset = dataset
-        self.split = split
-        self.revision = revision
-        self.streaming = streaming
-        self.data_files = data_files
+        self.config_kwargs = config_kwargs
+        if "split" not in self.config_kwargs:
+            warnings.warn("Default value of `split` will be changed to None in version 0.7", FutureWarning)
+        if "revision" not in self.config_kwargs:
+            warnings.warn("Default value of `revision` will be changed to None in version 0.7", FutureWarning)
```

Comment:
Looks like the default arguments are changed. @ejguan it will be slightly BC-breaking. WDYT?

Reply:
Wait, the default arguments remain the same right?
`split="train"`
`revision="main"`
`streaming=True`

The default arguments are assigned in `_get_response_from_huggingface_hub`.

Reply:
Oh in that case I'm fine with it, only that users will not be able to see those default arguments from IDE autocomplete. which is suboptimal but not a blocker. 

Is the warning for Streaming missing or we want it to stay `True`? The default is `False` for [HG's version](https://huggingface.co/docs/datasets/v2.8.0/en/package_reference/loading_methods#datasets.load_dataset).

Reply:
> Is the warning for Streaming missing or we want it to stay `True`?

Good question. I personally like streaming=True to incorporate the style of large-dataset.